pkg/kube/watcher.go

```diff
@@ -133,7 +132,7 @@ func WatchForServices(ctx context.Context, tracker *tcplistener.ListenerTracker,
 				continue
 			case event := <-eventCh:
 				if event.deleted {
-					if err := tracker.Remove(ctx, listenerAddr, int(event.port)); err != nil {
+					if err := tracker.Remove(ctx, net.ParseIP(*listenerAddr), int(event.port)); err != nil {
```

Comment:
Would it make sense to do the `net.ParseIP` once earlier and re-use the output?

Reply:
Okay, yes, I'll take a look.